agent/engine/task_manager.go

```diff
@@ -451,6 +473,50 @@ func (mtask *managedTask) startContainerTransitions(transitionFunc containerTran
 	return anyCanTransition, transitions
 }
 
+type containerTransitionFunc func(container *api.Container, nextStatus api.ContainerStatus)
+
+// containerNextState determines the next state a container should go to.
+// It returns: The state it should transition to, a bool indicating whether any
+// action is required, and a bool indicating whether a known status change is
+// possible.
+//
+// 'Stopped, false, true' -> "You can move it to known stopped, but you don't have to call a transition function"
+// 'Running, true, true' -> "You can move it to running and you need to call the transition function"
+// 'None, false, false' -> "This should not be moved; it has unresolved dependencies or is complete; no knownstatus change"
+//
+// Next status is determined by whether the known and desired statuses are
+// equal, the next numerically greater (iota-wise) status, and whether
+// dependencies are fully resolved.
+func (mtask *managedTask) containerNextState(container *api.Container) (api.ContainerStatus, bool, bool) {
+	clog := log.New("task", mtask.Task, "container", container)
```

Comment:
`clog` and `seelog` sounds like a confusing plan. Per my comment above, can we standardize on one at least where we're touching code?